src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -77,6 +78,29 @@ private static Exception GetInvalidArgumentException(string buffer, int offset,
             return ThrowHelper.GetArgumentException(ExceptionResource.Argument_InvalidOffsetLength);
         }
 
+        /// <summary>
+        /// Initializes an instance of the <see cref="StringSegment"/> struct.
+        /// </summary>
+        /// <param name="span">The original <see cref="Span{T}"/> used as buffer.</param>
+        public unsafe StringSegment(Span<char> span)
```

Comment:
I'm not sure this direction is very useful as it allocates which isn't really the point here. Going from StringSegment -> Span\<char\> or ReadOnlyMemory<char> makes more sense than this direction. I'd get rid of this ctor.

Reply:
I reverted the commit that adds the ctors, updated the PR.